app/controllers/orders_controller.rb

```diff
@@ -66,17 +72,25 @@ def destroy
     # Use callbacks to share common setup or constraints between actions.
     def set_order
       @order = Order.find(params[:id])
-      
+
     end
 
     def set_order_for_Product
       @product = Product.find(params[:product_id])
-    end 
+    end
 
     # Never trust parameters from the scary internet, only allow the white list through.
     def order_params
-      params.require(:order).permit(:order_number, :order_date, :order_total, :order_type,:product_id)
+      params.require(:order).permit(:order_number, :order_date, :order_total, :order_type,:product_id,:user_id, products_attributes: [:id])
     end
-    def order_params_for_product
-    end 
+    def user_required
+      if logged_in?
```

Comment:
There is no any business logic inside if loop

Please remove it

Reply:
i used this business logic for only logged in users because to access orders page user must logged in !!